packages/@aws-cdk/aws-amplify/lib/asset-deployment-handler/handler.ts

```diff
@@ -0,0 +1,137 @@
+// aws-sdk available at runtime for lambdas
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Amplify, S3 } from 'aws-sdk';
+import { AmplifyJobId, IsCompleteResponse, ResourceEvent, ResourceHandler } from './common';
+
+export interface AmplifyAssetDeploymentProps {
+  AppId: string;
+  BranchName: string;
+  S3BucketName: string;
+  S3ObjectKey: string;
+  TimeoutSeconds: number;
+}
+
+export class AmplifyAssetDeploymentHandler extends ResourceHandler {
+  private readonly props: AmplifyAssetDeploymentProps;
+  protected readonly amplify: Amplify;
+  protected readonly s3: S3;
+
+  constructor(amplify: Amplify, s3: S3, event: ResourceEvent) {
+    super(event);
+
+    this.props = parseProps(this.event.ResourceProperties);
+    this.amplify = amplify;
+    this.s3 = s3;
+  }
+
+  // ------
+  // CREATE
+  // ------
+
+  protected async onCreate(): Promise<AmplifyJobId> {
+    // eslint-disable-next-line no-console
+    console.log('deploying to Amplify with options:', JSON.stringify(this.props, undefined, 2));
+
+    // Verify no jobs are currently running.
+    const jobs = await this.amplify
+      .listJobs({
+        appId: this.props.AppId,
+        branchName: this.props.BranchName,
+        maxResults: 1,
+      })
+      .promise();
+
+    if (
+      jobs.jobSummaries &&
+      jobs.jobSummaries.length > 0 &&
+      jobs.jobSummaries[0].status == 'PENDING'
+    ) {
+      return Promise.reject('Amplify job already running. Aborting deployment.');
```

Comment:
@samkio  Maybe consider that some use case may deploy twice in short time (like mistake trigger or else)
I suggest use jobs.push(new job) instead reject.

Reply:
Thanks for taking a look @ChenKuanSun ; I understand that a deploy may happen multiple times in quick succession.

How should this be handled? Should the handler poll for X amount of time until there are no other jobs running? I'm not sure Amplify has a way of pushing a deployment job to a queue here?

Reply:
@samkio  
According this document:
**ListJobs**
https://docs.aws.amazon.com/amplify/latest/APIReference/API_ListJobs.html

It's Jobs array, and may have more than 100 Jobs, and maybe index 0 will be SUCCEED but second job is PENDING.

This line will ignore another excluded index 0

There is not Jobs.push(), Maybe you can try **StartJob**
https://docs.aws.amazon.com/amplify/latest/APIReference/API_StartJob.html

Reply:
right I see; so we should check all entries in the list are not in a pending state not just the first; makes sense I can make that change.

As for StartJob that is for the git repo sourced Amplify apps; it doesn't work for manually deployed apps (and hence we use StartDeployment which does: https://docs.aws.amazon.com/amplify/latest/APIReference/API_StartDeployment.html )

Reply:
Ok I have not used s3 deployment before so you just choice API which works.
Let's move forward when you change jobs condition.😉

Reply:
Change made :)